### PR TITLE
RSC15d

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

File: Spec/RestClient.swift [hunk 1]
```diff
@@ -485,6 +485,63 @@ class RestClient: QuickSpec {
...
+                        }
+                    }
+                }
+
+                // RSC15d
+                it("should not use an alternative host when the client receives a bad request") {
+                    let options = ARTClientOptions(key: "xxxx:xxxx")
+                    let client = ARTRest(options: options)
+                    client.httpExecutor = testHTTPExecutor
+                    testHTTPExecutor.http = MockHTTP(network: .HostBadRequest)
```
Comment (mattheworiordan) [inline Spec/RestClient.swift]: What is the definition of a bad request?

Comment (mattheworiordan) [inline Spec/RestClient.swift]: Ignore me, I see below... that is fine. Perhaps update the name to HostBadRequest400 to make it clearer for dummies like me :)

**Comment by tcard** [comment]
PTAL

**Comment by ricardopereira** [comment]
LGTM

**Pull request merged**